src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -506,7 +508,7 @@ else if (remainingNanos < 1_000_000_000) {
     // will wait for any previous attempt to complete, using the reader.stop and
     // writer.stop
     void tryToConnect(NatsUri cur, NatsUri resolved, long now) {
-        currentServer = null;
+        clearCurrentServer();

```

Comment:
clearCurrentServer sets current server to null after setting the lastServer variable, which helps reporting statuses during re/connect when current server needs to be null.